recyclerviewfastscroller/src/main/java/com/qtalk/recyclerviewfastscroller/RecyclerViewFastScroller.kt

```diff
@@ -343,23 +383,31 @@ class RecyclerViewFastScroller @JvmOverloads constructor(
                         //
 
                         // subtract the handle height offset
-                        val handleHeightOffset = handleImageView.height / 2
+                        val handleOffset = handleLength / 2
 
-                        val currentRelativeYPos =
-                            motionEvent.rawY - yAbsPosition - handleHeightOffset
+                        val currentRelativePos = when (fastScrollDirection) {
+                            FastScrollDirection.HORIZONTAL ->
+                                motionEvent.rawX - xAbsPosition - handleOffset
+                            FastScrollDirection.VERTICAL ->
+                                motionEvent.rawY - yAbsPosition - handleOffset
+                        }
 
                         // move the handle only if fastScrolled, else leave the translation of the handle to the onScrolled method on the listener
 
                         if (isFastScrollEnabled) {
-                            moveViewByRelativeYInBounds(handleImageView, currentRelativeYPos)
-                            moveViewByRelativeYInBounds(
+                            moveViewByRelativeInBounds(handleImageView, currentRelativePos)
```

Comment:
the name of this function seems a bit ambiguous, let change it to: 
```suggestion
                            moveViewByRelativeDirectionInBounds(handleImageView, currentRelativePos)
```

Reply:
Alrighty 👍

Reply:
@shahsurajk how about `moveViewToRelativePositionWithBounds`? What it really does is that it moves a view relative to its parent.

Reply:
can do. works!